libraries/abstractions/common_io/include/iot_uart.h

```diff
@@ -116,23 +117,32 @@ typedef struct
 } IotUARTConfig_t;
 
 /**
- * @brief Initializes the UART peripheral of the board with default configuration.
+ * @brief Initializes the UART peripheral of the board.
  *
  * The application should call this function to initialize the desired UART port.
  *
+ * @warning Once opened, the same UART instance must be closed before calling open again.
+ *
  * @param[in] lUartInstance The instance of the UART port to initialize.
  *
- * @return handle to the UART if everything succeeds, otherwise NULL.
+ * @return
+ * - 'the handle to the UART port (not NULL)', on success.
+ * - 'NULL', if
+ *     - invalid instance number
+ *     - open same instance more than once before closing it
  */
 IotUARTHandle_t iot_uart_open( int32_t lUartInstance );
 
 /**
  * @brief Sets the application callback to be called on completion of an operation.
  *
- * On completion of receive or transmit on UART, the application is notified with a
- * function callback with the state of the transaction. The callback function and the parameters
- * for the state of the callback are set using iot_uart_set_callback.
- * The application should call this function with the callback function and its parameters.
+ * The callback is guaranteed to be invoked when the current asynchronous operation completes, either successful or failed.
+ * This simply provides a notification mechanism to user's application. It has no impact if the callback is not set.
+ *
+ * @note This callback will not be invoked when synchronous operation completes.
+ * @note This callback is per handle. Each instance has its own callback.
+ * @note Single callback is used for both read_async and write_async. Newly set callback overrides the one previously set.
+ * @warning If the input handle is invalid, this function silently takes no action.
```

Comment:
Something I believe we should do for all common-io callback functions: We should specify if the user can pass in a  null callback, and if they can pass in a null user context. IMO they should be able to do both. Thoughts?

Reply:
Yes, they can do both. That is why I say "If the input handle is invalid, this function silently takes no action."

Do you have better suggestion?

Reply:
I think the other two inputs should be able to be set to null is what I mean

Reply:
I see. Will update.